src/Components.js

```diff
@@ -3,19 +3,32 @@ import DeleteButton from "./DeleteButton";
 
 const Components = () => {
     const [components, setComponents] = useState(null);
+    const [loading, setLoading] = useState(true);
+    const [error, setError] = useState(null);
 
     const [course, setCourse] = useState("> Frontend <");
 
     useEffect(() => {
-        fetch("http://localhost:8000/courses")
-        .then((response)=>{
-            return response.json()
-        })
-        .then((componentArray)=>{
-            console.log(componentArray)
-            setComponents(componentArray)
-        })
-    }, []); //dependency array
+        setTimeout(()=>{
+            fetch("http://localhost:8000/coursess")
+            .then((response)=>{
+                if (!response.ok){
+                    throw Error("Ooooooooopss! ...unable to connect to the server. >>>Check your internet connection and try again!")
+                }
+                return response.json()
+            })
+            .then((item)=>{
+                console.log(item)
+                setComponents(item)
+                setLoading(false)
+            })
+            .catch((error)=>{
+                console.log(error.message)
+                setError(error.message)
+                setLoading(false)
+            })
+        }, 5000)
+    }, []); //empty dependency array - allows the app to run only once
 
     const handleDelete = (id) => {
         console.log('The id of the clicked component is', id)
```

Comment:
Hello @Craftsman8.

Weldone on adding the features requested in this project. You are making a significant progress. Could you remove the delete functionality as we are not using it right now? 

Just so you know, removing the delete functionality will break the app if you don't remove all the references to that particular function immediately after removing the delete function.

Reply:
Kindly remove the unused students' array in the db.json file. as it is not needed at the moment.

Reply:
Kindly ensure that your app is running before you resubmit the link as it is not running currently. It is breaking.

Reply:
I'll do as you've instructed sir.. 🙇
I actually added an error to the server link so I can see my error message, but I forgot to remove it before submission